zerver/lib/actions.py

```diff
@@ -2229,120 +2229,23 @@ def do_create_realm(string_id, name, restricted_to_domain=None,
-    if log:
-        log_event(event)
-    send_event(event, [user_profile.id])
-
-
-def do_change_enable_sounds(user_profile, enable_sounds, log=True):
-    # type: (UserProfile, bool, bool) -> None
-    user_profile.enable_sounds = enable_sounds
-    user_profile.save(update_fields=["enable_sounds"])
-    event = {'type': 'update_global_notifications',
-             'user': user_profile.email,
-             'notification_name': 'enable_sounds',
-             'setting': enable_sounds}
-    if log:
-        log_event(event)
-    send_event(event, [user_profile.id])
-
-def do_change_enable_offline_email_notifications(user_profile, offline_email_notifications, log=True):
-    # type: (UserProfile, bool, bool) -> None
-    user_profile.enable_offline_email_notifications = offline_email_notifications
-    user_profile.save(update_fields=["enable_offline_email_notifications"])
-    event = {'type': 'update_global_notifications',
-             'user': user_profile.email,
-             'notification_name': 'enable_offline_email_notifications',
-             'setting': offline_email_notifications}
-    if log:
-        log_event(event)
-    send_event(event, [user_profile.id])
-
-def do_change_enable_offline_push_notifications(user_profile, offline_push_notifications, log=True):
-    # type: (UserProfile, bool, bool) -> None
-    user_profile.enable_offline_push_notifications = offline_push_notifications
-    user_profile.save(update_fields=["enable_offline_push_notifications"])
-    event = {'type': 'update_global_notifications',
-             'user': user_profile.email,
-             'notification_name': 'enable_offline_push_notifications',
-             'setting': offline_push_notifications}
-    if log:
-        log_event(event)
-    send_event(event, [user_profile.id])
-
-def do_change_enable_online_push_notifications(user_profile, enable_online_push_notifications, log=True):
-    # type: (UserProfile, bool, bool) -> None
-    user_profile.enable_online_push_notifications = enable_online_push_notifications
-    user_profile.save(update_fields=["enable_online_push_notifications"])
-    event = {'type': 'update_global_notifications',
-             'user': user_profile.email,
-             'notification_name': 'enable_online_push_notifications',
-             'setting': enable_online_push_notifications}
-    if log:
-        log_event(event)
-    send_event(event, [user_profile.id])
-
-def do_change_enable_digest_emails(user_profile, enable_digest_emails, log=True):
-    # type: (UserProfile, bool, bool) -> None
-    user_profile.enable_digest_emails = enable_digest_emails
-    user_profile.save(update_fields=["enable_digest_emails"])
+def do_change_notification_settings(user_profile, name, value, log=True):
+    # type: (UserProfile, str, bool, bool) -> None
+    """Takes in a UserProfile object, the name of a global notification
+    preference to update, and the value to update to
+    """
 
-    if not enable_digest_emails:
```

Comment:
Yep you're totally right. I'm fine with the magic string inside the `do_change_notification_settings` method, or we can chat in person if you want to discuss a cleaner possible solution